### PR TITLE
feat: add aave strategy

### PR DESCRIPTION
**Pull request by toprince**

### CODE REVIEW

File: contracts/earn/strategies/Aave/StrategyAave.sol [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+        // claim all rewards to the vault
+        IAaveV3Incentives(incentivesController).claimAllRewards(rewardAssets, address(vault));
+
+        uint256 wantHarvested = balanceOfWant() - beforeBal;
+        uint256 fee = chargePerformanceFee(wantHarvested);
+        deposit();
+
+        lastHarvest = block.timestamp;
+        emit StratHarvest(msg.sender, wantHarvested - fee, balanceOf());
```
Comment (fb-alexcq) [inline contracts/earn/strategies/Aave/StrategyAave.sol:58]: No need to calculate the balance, charge perf fee and deposit.

File: contracts/earn/strategies/Aave/StrategyAave.sol [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+    /* ----- Public Functions ----- */
+
+    function harvest() public override whenNotPaused {
+        require(msg.sender == tx.origin || msg.sender == address(vault), "StrategyAave: EOA_or_vault");
+
+        uint256 beforeBal = balanceOfWant();
+
+        // claim all rewards to the vault
+        IAaveV3Incentives(incentivesController).claimAllRewards(rewardAssets, address(vault));
```
Comment (fb-alexcq) [inline contracts/earn/strategies/Aave/StrategyAave.sol:51]: This API supports directly claiming into the specified address

**Comment by fb-alexcq** [comment]
```
npm run "format:write"
```

File: contracts/earn/strategies/Aave/StrategyAave.sol [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    /* ----- State Variables ----- */
+
+    address[] public rewardAssets;
+    uint256 public lastHarvest;
+
+    /* ----- Constant Variables ----- */
+
+    address public constant aavePool = address(0x794a61358D6845594F94dc1DB02A252b5b4814aD); // Aave: Pool V3
+    address public constant incentivesController = address(0x929EC64c34a17401F460460D4B9390518E5B473e); // Aave: Incentives V3
+    address public constant dataProvider = address(0x69FA688f1Dc47d4B5d8029D5a35FB7a548310654); // Aave: Pool Data Provider V3
```
Comment (fb-alexcq) [inline contracts/earn/strategies/Aave/StrategyAave.sol:27]: @0xmer1in  could you please help double check these addresses in deployment?

Comment (0xmer1in) [reply contracts/earn/strategies/Aave/StrategyAave.sol:27]: both correct

**Review by 0xmer1in — APPROVED** [review]

**Review by fb-alexcq — APPROVED** [review]

**Pull request merged**